### PR TITLE
add auto learning rate scaling for `TorchNano`

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add auto learning rate scaling for `TorchNano`, and add an UT for it

### 1. Why the change?

To align the arguments of `TorchNano` and `Trainer`

### 2. User API changes

Now `TorchNano` has a new argument `auto_lr` to control whether to scale learning rate automatially when using multi-instance training.

### 3. Summary of the change 

Add auto learning rate scaling for `TorchNano`, and add an UT for it

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

**Comment by yangw1234** [comment]
LGTM
